packages/components/organisms/f-checkout/src/store/checkoutAnalytics.module.js

```diff
@@ -78,20 +78,18 @@ export default {
          * Pushes `form` event to the dataLayer with correct data
          */
         trackFormInteraction ({ state, rootState }, { action, error }) {
-            const formName = rootState.checkout.isLoggedIn ? 'checkout' : 'checkout_guest';
+            const formName = rootState[VUEX_CHECKOUT_MODULE].isLoggedIn ? 'checkout' : 'checkout_guest';
 
             const mappedError = error ? mapAnalyticsNames(error) : null;
 
-            Trak.event({
+            window.dataLayer.push({
```

Comment:
Not sure if I missed the conversation  – why are we switching to dataLayer calls directly in the code?

Would be better to abstract this as otherwise we end up with 3rd party specific API calls (i.e. GA dataLayer calls) throughout our code. Main reason we try to avoid this is because Google has a history of changing the API structure of these calls and that makes them every brittle.

If we abstract these calls, we only have to change the call to dataLayer in one place rather than lots of places.

Reply:
Hey, so `f-trak` doesn't work in CoreWeb, @tommcclean had a look into it yesterday and found that a dependency, way down the line that is doesn't work on isomorphic web. 😬

Reply:
Ok, makes sense – in that case, we should create some tickets to fix `f-trak` so that it's isomorphic.

Don't mind this going in as a temporary solution, but we should make sure we fix the root problem too otherwise it'll come back to bite us later.